src/transport/manager/mod.rs

```diff
@@ -170,27 +170,31 @@ impl Stream for TransportContext {
     type Item = (SupportedTransport, TransportEvent);
 
     fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
-        let len = match self.transports.len() {
-            0 => return Poll::Ready(None),
-            len => len,
-        };
-        let start_index = self.index;
-
-        loop {
-            let index = self.index % len;
-            self.index += 1;
+        if self.transports.is_empty() {
+            // Terminate if we don't have any transports installed.
+            return Poll::Ready(None);
+        }
 
-            let (key, stream) = self.transports.get_index_mut(index).expect("transport to exist");
+        let len = self.transports.len();
+        for index in 0..len {
+            let current = (self.index + index) % len;
```

Comment:
nit: what about updating `self.index` before the loop to not care about it in every return branch? In the end, it doesn't matter if we store the last index or the new one between the `poll_next` calls.

```suggestion
        let len = self.transports.len();
        self.index = (self.index + 1) % len;
        for index in 0..len {
            let current = (self.index + index) % len;
```